MekHQ/src/mekhq/campaign/LogEntry.java

```diff
@@ -35,116 +35,124 @@
-    public void setDate(Date d) {
-        this.date = d;
-    }
-    
-    public Date getDate() {
-        return date;
-    }
-
-    public String getDateString() {
-        return date.toString();
-    }
-
-    public void setDesc(String d) {
-        this.desc = Objects.requireNonNull(d);
-    }
-    
-    public String getDesc() {
-        return desc;
-    }
-    
-    public void setType(String type) {
-        this.type = type;
-    }
-    
-    public String getType() {
-        return type;
-    }
-    
-    public boolean isType(String type) {
-        return Objects.equals(this.type, type);
-    }
-    
-    @Override
-    public void writeToXml(PrintWriter pw1, int indent) {
-        StringBuilder sb = new StringBuilder();
-        sb.append(MekHqXmlUtil.indentStr(indent)).append("<logEntry>");
-        if(null != date) {
-            sb.append("<date>").append(DATE_FORMAT.format(date)).append("</date>");
-        }
-        sb.append("<desc>").append(MekHqXmlUtil.escape(desc)).append("</desc>");
-        if(null != type) {
-            sb.append("<type>").append(MekHqXmlUtil.escape(type)).append("</type>");
-        }
-        sb.append("</logEntry>");
-        pw1.println(sb.toString());
-    }
-    
-    public static LogEntry generateInstanceFromXML(Node wn) {
-        final String METHOD_NAME = "generateInstanceFromXML(Node)"; //$NON-NLS-1$
-
-        LogEntry retVal = new LogEntry();
-        
-        try {    
-            // Okay, now load fields!
-            NodeList nl = wn.getChildNodes();
-            
-            for (int x=0; x<nl.getLength(); x++) {
-                Node wn2 = nl.item(x);
-                
-                if (wn2.getNodeName().equalsIgnoreCase("desc")) {
-                    retVal.desc = wn2.getTextContent();
-                } else if (wn2.getNodeName().equalsIgnoreCase("type")) {
-                    retVal.type = wn2.getTextContent();
-                } else if (wn2.getNodeName().equalsIgnoreCase("date")) {
-                    retVal.date = DATE_FORMAT.parse(wn2.getTextContent().trim());
-                }
-            }
-        } catch (Exception ex) {
-            // Doh!
-            MekHQ.getLogger().log(LogEntry.class, METHOD_NAME, ex);
-        }
-        
-        return retVal;
-    }
-    
-    @Override
-    public String toString() {
-        StringBuilder sb = new StringBuilder();
-        if(null != date) {
-            sb.append("[").append(DATE_FORMAT.format(date)).append("] ");
-        }
-        sb.append(desc);
-        return sb.toString();
-    }
-    
-    @Override
-    public LogEntry clone() {
-        return new LogEntry(getDate(), getDesc(), getType());
-    }
-}
\ No newline at end of file
+public class LogEntry implements Cloneable, MekHqXmlSerializable {
+
+	private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss"); //$NON-NLS-1$
+	// LATER SimpleDateFormat is not thread-safe: use throw-away instances or switch to java.time
```

Comment:
You could go ahead and implement this now, transitioning `DATE_FORMAT` to a `String` and building the `SimpleDateFormat` on the fly below.